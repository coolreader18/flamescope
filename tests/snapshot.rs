#[test]
fn test_basic() {
    flame::clear();
    {
        let _guard = flame::start_guard("the main");
        {
            let _guard_two = flame::start_guard("foobar");
        }
    }
    let actual = flamescope::spans_to_speedscope(flame::spans());
    insta::assert_json_snapshot!(actual, {
        ".profiles[].events[].at" => 100,
        ".profiles[].startValue" => 50,
        ".profiles[].endValue" => 150,
    });
}
