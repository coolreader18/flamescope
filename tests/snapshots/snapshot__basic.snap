---
source: tests/snapshot.rs
expression: actual

---
{
  "$schema": "https://www.speedscope.app/file-format-schema.json",
  "profiles": [
    {
      "type": "evented",
      "name": "the main",
      "unit": "nanoseconds",
      "startValue": 50,
      "endValue": 150,
      "events": [
        {
          "type": "O",
          "at": 100,
          "frame": 0
        },
        {
          "type": "O",
          "at": 100,
          "frame": 1
        },
        {
          "type": "C",
          "at": 100,
          "frame": 1
        },
        {
          "type": "C",
          "at": 100,
          "frame": 0
        }
      ]
    }
  ],
  "shared": {
    "frames": [
      {
        "name": "the main",
        "file": null,
        "line": null,
        "col": null
      },
      {
        "name": "foobar",
        "file": null,
        "line": null,
        "col": null
      }
    ]
  },
  "activeProfileIndex": null,
  "exporter": null,
  "name": null
}
